src/main/java/edu/ie3/datamodel/models/value/WeatherValue.java

```diff
@@ -27,44 +28,100 @@ public class WeatherValue implements Value {
     this.coordinate = coordinate;
     this.solarIrradiance = solarIrradiance;
     this.temperature = temperature;
     this.wind = wind;
+    this.groundTemperature0cm = groundTemperature0cm;
+    this.groundTemperature80cm = groundTemperature80cm;
   }
 
   /**
+   * Constructor with all parameters as quantities.
+   *
    * @param coordinate of this weather value set
    * @param directSolarIrradiance Direct sun irradiance for this coordinate (typically in W/m²)
    * @param diffuseSolarIrradiance Diffuse sun irradiance for this coordinate (typically in W/m²)
    * @param temperature for this coordinate (typically in K)
    * @param direction Direction, the wind comes from as an angle from north increasing clockwise
    *     (typically in rad)
    * @param velocity Wind velocity for this coordinate (typically in m/s)
+   * @param groundTemp0cm Ground temperature at 0cm for this coordinate (typically in K, can be
+   *     null)
+   * @param groundTemp80cm Ground temperature at 80cm for this coordinate (typically in K, can be
+   *     null)
    */
   public WeatherValue(
       Point coordinate,
       ComparableQuantity<Irradiance> directSolarIrradiance,
       ComparableQuantity<Irradiance> diffuseSolarIrradiance,
       ComparableQuantity<Temperature> temperature,
       ComparableQuantity<Angle> direction,
-      ComparableQuantity<Speed> velocity) {
+      ComparableQuantity<Speed> velocity,
+      ComparableQuantity<Temperature> groundTemp0cm,
+      ComparableQuantity<Temperature> groundTemp80cm) {
     this(
         coordinate,
         new SolarIrradianceValue(directSolarIrradiance, diffuseSolarIrradiance),
         new TemperatureValue(temperature),
-        new WindValue(direction, velocity));
+        new WindValue(direction, velocity),
+        groundTemp0cm == null ? null : new GroundTemperatureValue(groundTemp0cm),
+        groundTemp80cm == null ? null : new GroundTemperatureValue(groundTemp80cm));
+  }
+
+  /**
+   * @deprecated Use the constructor that includes ground temperatures instead
+   */
+  @Deprecated(since = "3.0", forRemoval = true)
+  public WeatherValue(
+      Point coordinate,
+      SolarIrradianceValue solarIrradiance,
+      TemperatureValue temperature,
+      WindValue wind) {
+    this(coordinate, solarIrradiance, temperature, wind, null, null);
+  }
+
+  /**
+   * @deprecated Use the constructor that includes ground temperatures instead
+   */
+  @Deprecated(since = "3.0", forRemoval = true)
+  public WeatherValue(
+      Point coordinate,
+      ComparableQuantity<Irradiance> directSolarIrradiance,
+      ComparableQuantity<Irradiance> diffuseSolarIrradiance,
+      ComparableQuantity<Temperature> temperature,
+      ComparableQuantity<Angle> direction,
+      ComparableQuantity<Speed> velocity) {
+    this(
+        coordinate,
+        directSolarIrradiance,
+        diffuseSolarIrradiance,
+        temperature,
+        direction,
+        velocity,
+        null,
+        null);
```

Comment:
This nice, but as I remember right, we no longer care for backwards compability. If you remove this, there might be more things that no longer are necessary.